### PR TITLE
TST: Ensure hdr plot tests always use fresh data

### PR DESCRIPTION
**Pull request by bashtage**
Avoid data reuse in hdr plots

### CODE REVIEW

**Comment by bashtage** [comment]
There are occasional failures in this on the random testing:

https://travis-ci.org/bashtage/statsmodels/jobs/426374967

File: statsmodels/graphics/tests/test_functional.py [hunk 1]
```diff
@@ -17,13 +17,20 @@
     have_matplotlib = False
 
 
-data = elnino.load(as_pandas=False)
-labels = data.raw_data[:, 0].astype(int)
-data = data.raw_data[:, 1:]
+@pytest.fixture()
```
Comment (jbrockmendel) [inline statsmodels/graphics/tests/test_functional.py]: Will these be useful elsewhere?  i.e. would it be worthwhile to put them in a conftest?

Comment (bashtage) [reply statsmodels/graphics/tests/test_functional.py]: I these are specific enough t this module to not warrant ending up in conftest.

**Comment by josef-pkt** [comment]
maybe there is a bug in hdr plot
Is the `data` modified? or why do we need to reload the data each time?

**Comment by bashtage** [comment]
I assume there is a subtle bug but it is hard to find since it only rarely errors.

**Comment by bashtage** [comment]
If this is merged, then if there are still random failures this would indicate an algorithm issue (or something deeper, like BLAS)

**Comment by bashtage** [comment]
This is not the issue.  It is due to using the global RandomState internally by hdr plot

**Comment by josef-pkt** [comment]
Is the randomness in differential_evolution?
I don't see any direct use of random numbers.

**Comment by josef-pkt** [comment]
ok, just saw comment in #5148 that sets seed for differential evolution

**Pull request closed without merging**